### PR TITLE
Keyword overrides (header information)

### PR DESCRIPTION
**Pull request by RinconResearch**
When plotting a bluefile using overlay_array or show_array, the header contents have been ignored.  Without xstart and xdelta, all files are plotted starting at the zero point and incrementing at the same rate.  This can be seen in the notebook dataplot-before.ipynb from the example directory when run against the master branch.

I changed the input structure to be both data *and* header and modified overlay_array and show_array.  The results can be seen by checking out the kwoverrides branch and running the notebook dataplot-after.ipynb in the example directory.

Note: you'll need bluefile.py, which is not included in this repository.

### CODE REVIEW

File: jupyter_sigplot/sigplot.py [hunk 1]
```diff
@@ -217,6 +188,42 @@ def overlay_file(self, path):
         self.overlay_href(path)
 
 
+
+def _prepare_array_input(input, layer_type, subsize):
```
Comment (mrecachinas) [inline jupyter_sigplot/sigplot.py]: Thanks for popping this out into another function, much more modular!

You'll probably want to write a test or two against this function as well as fix the tests that broke due to this function.

File: jupyter_sigplot/sigplot.py [hunk 2]
```diff
@@ -217,6 +188,42 @@ def overlay_file(self, path):
...
+    # TODO (sat 2018-11-08): I think this needs to move into a
+    # function so we can test it better. At that point, we may
+    # also be able to specify it as the serializer for the
+    # traitlet via the `to_json` keyword argument
+    # TODO (ddw 2018-12-10): Do I need to check the number of
+    # items in input?
+    data, header = input
```
Comment (mrecachinas) [inline jupyter_sigplot/sigplot.py]: You shouldn't need to check the number of items in input as long as you ensure `input` is always a tuple consisting of `(data, header)` (with the type `Tuple[List, Dict]`).

File: jupyter_sigplot/sigplot.py [hunk 3]
```diff
@@ -217,6 +188,42 @@ def overlay_file(self, path):
...
 
 
+
+def _prepare_array_input(input, layer_type, subsize):
+    # TODO (sat 2018-11-08): I think this needs to move into a
+    # function so we can test it better. At that point, we may
+    # also be able to specify it as the serializer for the
+    # traitlet via the `to_json` keyword argument
+    # TODO (ddw 2018-12-10): Do I need to check the number of
+    # items in input?
```
Comment (mrecachinas) [inline jupyter_sigplot/sigplot.py]: Probably also should add a docstring here with some details about the function and the arguments.

**Comment by mrecachinas** [comment]
@RinconResearch I posted a few comments where once addressed I'll gladly merge this fix in!

File: jupyter_sigplot/sigplot.py [hunk 4]
```diff
@@ -63,8 +63,7 @@ def __init__(self, *args, **kwargs):
     def change_settings(self, **kwargs):
         self.options.update(kwargs)
 
-    def show_array(self, data, layer_type="1D", subsize=None):
-        overrides = {}
+    def show_array(self, data, overrides = {}, layer_type="1D", subsize=None):
```
[the remark this answers is not in the document]
Comment (RinconResearch) [reply jupyter_sigplot/sigplot.py]: Thanks for pointing this out.  I've changed it in my code and will use it in the future!

**Pull request closed without merging**